ibis/backends/sql/rewrites.py

```diff
@@ -100,10 +122,23 @@ def sort_to_select(_, **kwargs):
 
 @replace(p.WindowFunction)
 def window_function_to_window(_, **kwargs):
-    """Convert a WindowFunction node to a Window node."""
+    """Convert a WindowFunction node to a Window node.
+
+    Also rewrites first -> first_value, last -> last_value.
+    """
+    func = _.func
+    if isinstance(func, (ops.First, ops.Last)):
+        if func.where is not None:
```

Comment:
This doesn't seem conceptually invalid. Perhaps we should avoid failing here so that if/when this is implemented it's an `XPASS`.

Reply:
It's not clear how that would work - the issue here is that `first()`/`last()` in ibis can serve as both reductions and analytic functions, where in the windowing context they only work as analytic functions (no `where` clause available). Since this isn't a regression from 8.0 (or from the current behavior) I'm inclined to not figure out how to plumb `where` through here and get it to fail nicely until/if compilation works later.